nmagent/responses_test.go

```diff
@@ -0,0 +1,137 @@
+package nmagent_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/Azure/azure-container-networking/nmagent"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestContainsFixes(t *testing.T) {
+	tests := []struct {
+		name  string
+		resp  nmagent.AzResponse
+		fixes []nmagent.HomeAZFix
+		exp   bool
+	}{
+		{
+			"empty",
+			nmagent.AzResponse{},
+			[]nmagent.HomeAZFix{},
+			true,
+		},
+		{
+			"one present",
+			nmagent.AzResponse{
+				AppliedFixes: []nmagent.HomeAZFix{
+					nmagent.HomeAZFixIPv6,
+				},
+			},
+			[]nmagent.HomeAZFix{nmagent.HomeAZFixIPv6},
+			true,
+		},
+		{
+			"one absent",
+			nmagent.AzResponse{
+				AppliedFixes: []nmagent.HomeAZFix{},
+			},
+			[]nmagent.HomeAZFix{nmagent.HomeAZFixIPv6},
+			false,
+		},
+		{
+			"one with empty request",
+			nmagent.AzResponse{
+				AppliedFixes: []nmagent.HomeAZFix{
+					nmagent.HomeAZFixIPv6,
+				},
+			},
+			[]nmagent.HomeAZFix{},
+			true,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			got := test.resp.ContainsFixes(test.fixes...)
+
+			exp := test.exp
+			if got != exp {
+				t.Error("unexpected response from ContainsFixes: exp:", exp, "got:", got)
```

Comment:
Just for my understanding. Is there a strong reason to use Testify package's _**.Error**_ and _**.Fatal**_ over require package? IMO require is more succinct and readable.

Reply:
We don't need to wait for this. Resolving.

Reply:
I find `require` and `assert` to be obtuse, and they obscure simple things like "which is the expected value and the one that you received." It's easy to accidentally swap them and get confusing output in test failures. Using `testing` lets me have better control over the failure output so that I can make sure that it's useful to whoever is reading it (usually me, but not always). Also, `testing.(*T).Error` and `testing.(*T).Fatal` are part of the standard library and have a small API surface area that you can keep in your head. `require` and `assert` have an enormous number of functions to assert everything under the sun... that I usually never need. If I do routinely need to perform some complex assertion, I just create a helper with `testing.(*T).Helper` and use that. This is good enough to test the entire standard library, and it's good enough for me.

If I choose to use an assertion package on a project, I much prefer the lightweight https://github.com/matryer/is than testify.

Anyway, I realize others like `testify` and don't share my opinion of it. Given that, I only ask people to pick one and stick with it (don't mix styles in the same test) and leave the choice to them.

Reply:
Thanks for the explanation, Tim. Sounds good.